### PR TITLE
Clip embedding

### PR DESCRIPTION
**Pull request by bryanyzhu**
*Issue #, if available:*
NA

*Description of changes:*
Add a tutorial of using CLIP to extract embeddings for retrieval problem.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/tutorials/multimodal/clip_embedding.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+Let's extract some image embedding from the CLIP vision encoder,
+
+```{.python .input}
+from autogluon.multimodal import MultiModalPredictor
+predictor = MultiModalPredictor(hyperparameters={"model.names": ["clip"]}, problem_type="zero_shot")
+
+# extract image embeddings.
+image_embeddings = predictor.extract_embedding({"image": image_paths})
+for k, v in image_embeddings.items():
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:40]: Since we know there is only one key `image` here, we may just use `image_embeddings["image"]`.

File: docs/tutorials/multimodal/clip_embedding.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+```
+
+The output has a shape of (5, 768), because there are 5 images, each of the image embedding has a dimension of 768. 
+
+Similarly, you can also extract text embeddings for other tasks, 
+
+```{.python .input}
+# extract text embeddings.
+text_embeddings = predictor.extract_embedding({"text": ['There is a carriage in the image']})
+for k, v in text_embeddings.items():
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:51]: Ditto.

File: docs/tutorials/multimodal/clip_embedding.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+text_features = text_embeddings['text']
+
+from numpy import linalg as LA
+import numpy as np
+image_features /= LA.norm(image_features, axis=-1, keepdims=True)
+text_features /= LA.norm(text_features, axis=-1, keepdims=True)
+
+similarity = 100.0 * np.matmul(image_features, text_features.T)
+from scipy.special import softmax
+print(softmax(similarity))
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:77]: Do we only need to do `argmax` instead of `softmax`?

File: docs/tutorials/multimodal/clip_embedding.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+If we want to switch to another text query, we simply re-compute text embeddings and do this similarity comparison again,
+
+```{.python .input}
+text_embeddings = predictor.extract_embedding({"text": ['There is an airplane over a car.']})
+
+text_features = text_embeddings['text']
+text_features /= LA.norm(text_features, axis=-1, keepdims=True)
+
+similarity = 100.0 * np.matmul(image_features, text_features.T)
+print(softmax(similarity))
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:97]: Ditto.

**Review by zhiqiangdon — APPROVED** [review]
LGTM. Thanks!

File: docs/tutorials/multimodal/clip_embedding.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+display(pil_img)
+```
+
+If we want to switch to another text query, we simply re-compute text embeddings and do this similarity comparison again,
+
+```{.python .input}
+text_embeddings = predictor.extract_embedding({"text": ['There is an airplane over a car.']})
+
+text_features = text_embeddings['text']
+text_features /= LA.norm(text_features, axis=-1, keepdims=True)
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:94]: No need to normalize the features, which are already normalized inside the predictor.

Comment (bryanyzhu) [reply docs/tutorials/multimodal/clip_embedding.md:94]: Cool, thanks

File: docs/tutorials/multimodal/clip_embedding.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+
+Let's reuse the example we have above. We already have 5 image embeddings in **image_embeddings**, and 1 text embedding in **text_embeddings**, now we normalize the feature and then compute their similarities,
+
+```{.python .input}
+image_features = image_embeddings['image']
+text_features = text_embeddings['text']
+
+import numpy as np
+
+similarity = 100.0 * np.matmul(image_features, text_features.T)
```
Comment (zhiqiangdon) [inline docs/tutorials/multimodal/clip_embedding.md:70]: Do we need to remove 100?

**Comment by bryanyzhu** [comment]
This PR is ready to be merged except Jenkins failure, feel free to provide comments, thank you.

**Review by gradientsky — APPROVED** [review]
